docs/src/references/mqtt-api.md

```diff
@@ -0,0 +1,836 @@
+where the `@parent` device info is derived from the topic itself, by default.
+But, it is advised to declare it explicitly as follows:
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/child01/service/nodered' '{
+  "@type": "service",
+  "@parent": "te/device/child01",
+  "displayName": "nodered",
+  "type": "systemd"
+}'
+```
+
+#### Register a service of a nested child device
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/nested_child01/service/nodered' '{
+  "@type": "service",
+  "@parent": "te/device/nested_child01",
+  "displayName": "nodered",
+  "type": "systemd"
+}'
+```
+
+### Auto Registration
+
+Users can use an auto-registration mechanism for immediate child devices and services,
+if they conform to the following topic scheme that clearly demarcates the `device` and `service` in the topic itself:
+
+```text
+te/device/<device_id>/service/<service_id>
+```
+
+Where the second subtopic level is `device` and 4th level is `service`.
+
+For example, if the following measurement message is received without any explicit registrations,
+
+```text
+te/device/rpi1001/service/collectd/m/cpu_usage
+```
+
+`rpi1001` and `collectd` will be auto-registered as `device` and `service` types.
+
+:::caution
+Auto-registration of entities can be enabled/disabled via tedge configuration.
+
+Users are highly encouraged to register the devices manually as it allows devices full control over their registration process. Meta information can also be added to the device to better describe the device's custom type and function.
+:::
+
+### Entity store
+
+All the entity registration messages retained with the MQTT broker helps thin-edge components to
+maintain a entity store with all the registered devices and services along with their metadata.
+
+Components like mappers use such an entity store while processing data from various sources.
+For example, a mapper component can get info about all the registered entities with a simple subscription
+for all possible entity identification topic levels as follows:
+
+```sh
+mosquitto_sub -t 'te/+' -t 'te/+/+' -t 'te/+/+/+' -t 'te/+/+/+/+'
+```
+
+:::note
+The MQTT broker is used as the persistence layer to store the registered entities (assuming the registration messages were published with the MQTT retain flag).
+:::
+
+## Data types
+
+Telemetry and commands use the data type topic levels after the entity/component subtopics.
+Even for the data type levels, a user is free to define those as they wish.
+But thin-edge has some pre-defined `<data-type>` subtopics for well-known types like
+*measurements*, *alarms*, *events* and *commands*, on which it enforces some constraints,
+so that it can process them.
+
+## Telemetry data
+
+|Type|Topic scheme|
+|------|-------|
+|Measurements|`te/<identifier>/m/[measurement-type]`|
+|Events|`te/<identifier>/e/<event-type>`|
+|Alarms|`te/<identifier>/a/<alarm-type>`|
+
+### Examples: With default device/service topic semantics
+
+#### Publish to the main device
```

Comment:
I guess depends which user's perspective we're taking. The reference docs are more for the developers which need to know how to publish telemetry data and assign it to the main device...

I guess I just find "Publish on behalf of the main device" too long for a title...though I do agree "on behalf of" is more accurate and less open to interpretation.

Reply:
How about titles like "Main device telemetry/data", "Child device data" etc without specifying any to/from.

Reply:
Actually since we are already in the "telemetry data" section we can drop that part as well...Let's just call it:

* Main device
* Child device

etc.

Reply:
The titles can stay as of now.